src/ServiceControl.Audit/Persistence/IFailedAuditStorage.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace ServiceControl.Audit.Persistence
+{
+    using System.Threading;
+    using System;
+    using System.Threading.Tasks;
+    using Auditing;
+
+    interface IFailedAuditStorage
+    {
+        Task Store(dynamic failure);
+
+        Task ProcessFailedMessages(
+            Func<FailedTransportMessage, Func<CancellationToken, Task>, CancellationToken, Task> onMessage,
```

Comment:
I wonder if it'd be better to have delegates defined for these and not rely on implicit `Func`. Especially in this case because we have `Func<T, Func<U, Task>, U, Task>`.